### DISCUSSION TITLE
debug guidance and learning instruction

### DISCUSSION THREAD

**Discussion by DMwangnima** [juicedata/juicefs#1048] [Q&A] [answered]
Hey guys! I'm a rookie for this project, so analyzing source codes directly would be difficult for me. I would appreciate it very much if anyone could tell me **how to debug syscalls like read() or write() using goland with having mounted _juicefs**_ and _**some learning instructions for this funny project**_.

**Comment by davies** [comment, marked as answer]
In order to debug read/write, you need to launch `juicefs mount xxx xxx` process ( without `-d` option) from goland, then you can add interrupt point and others.

btw, once the process is interrupted, all the access to the mount point will hang, so some of your applications may be not responsible.
